### ISSUE TITLE
Implement database dump and restore commands

### ISSUE THREAD

**Issue by philipmjohnson** [radgrad/radgrad#63]
Please do all work for this issue in the branch issue-63.

The dump command:
  * Produces one or more JSON files that are downloaded to the client. (If multiple files, then they must be zipped up prior to download.)
  * See the [ICS Syllabus Manager Export Command](https://github.com/uhm-ics/ics-syllabus#exporting-syllabus-data) for example code.
  * There are no docIDs in the dumped file; all entities are referred to by slug.
  * The result of the dump command is a listing on the page of the collections dumped and the total number of entries in each.  
  * Prior to doing the dump, the command performs an Integrity Check and does not do the dump if any problems are found.

The restore command:
  *  requires a developer to manually put the downloaded "dump" data file into the server/startup/restore directory.
  * On startup, if this directory exists, then the database is restored.  "Restore" means that all entities in the database are deleted, then the dumped file is loaded.

**Comment by philipmjohnson** [comment]
The "dump" part is now implemented.  

First, all collections (except for the MentorSpace collections) now implement a dumpOne(docID) method that produces an object, which, when passed to that collection's define() method, will recreate that entity.  

The Mentorspace collections need some slight redesign, which is why I did not implement dumpOne for them yet. 

BaseCollection implements a dumpAll() method that iterates through all the instances of that collection, and collects and returns the results of calling dumpOne() on each instance.

The UI implements a Dump Database command that invokes a method to run the dumpAll over all collections (except the MentorSpace ones), and returns the resulting array to the client.  This data is then:
  1. Displayed in the UI through accordions
  2. Zipped up and downloaded to the client. 

Here is a screenshot of a successful invocation of Dump Database:

![dumpdb-success-message](https://cloud.githubusercontent.com/assets/290288/21913708/3017f9ee-d8d2-11e6-863c-c8060f673eea.png)

If an error occurred, the UI reports it. Here's an example:

![dumpdb-error-message](https://cloud.githubusercontent.com/assets/290288/21913724/49e0db70-d8d2-11e6-858f-5e7616cc26ef.png)

Next steps:
* Fix MentorSpace collections and include them.
* Implement the ability to initialize the system from the downloaded file.
